### DISCUSSION TITLE
High io wait

### DISCUSSION THREAD

**Discussion by BlackWand1** [minio/minio#18900] [Q&A]
We have a pool with 4 servers, each with 12 drives. Even at low incomming of data the cpu "io wait" is almost maxxed out.
When checking "top" "htop" the cpu is more or less idle.

See io wait graph below. From what I understand this means that much time for CPU is "wasted" on waiting for I/O from/to drives.
But what if we get 12 more drives and add as one more pool (in same hardware). How would that work with the cpu being "maxxed out" on io wait?

or does that only relate to these processes? and with the new pool and their own processed would have their "own" io wait?

![image](https://github.com/minio/minio/assets/85252943/5ea07f03-8369-49e5-a843-cb6c65263ee1)

**Comment by BlackWand1** [comment]
Just to clarify, it is 4x12 normal spinning disk drives.

**Reply by klauspost** [reply]
@BlackWand1 We only perform performance analysis as part of our [subscriptions](https://min.io/pricing). However I am not sure how you would expect MinIO to solve iowait, since your hardware cannot keep up.

**Reply by BlackWand1** [reply]
A better understanding of what "io wait" means.
Since it is added to the "cpu utilization" it looks like cpu is working full time. But yet, we could add more incomming data and it swallows it good.

My guess is that it is just a way to show that those active processes/threads are busy with "io wait". But if new threads are started, the cpu still got time for them.

Maybe i should just remove "io wait" from the "cpu utilization" to get a better estimate.

**Comment by Jazz9** [comment]
What are you using it for?
